pinot-core/src/main/java/com/linkedin/pinot/core/operator/query/DictionaryBasedAggregationOperator.java

```diff
@@ -74,15 +76,27 @@ protected IntermediateResultsBlock getNextBlock() {
           AggregationFunctionFactory.AggregationFunctionType.valueOf(function.getName().toUpperCase());
       String column = aggregationFunctionContext.getAggregationColumns()[0];
       Dictionary dictionary = _dictionaryMap.get(column);
-      AggregationResultHolder resultHolder;
+      FieldSpec.DataType dataType = _dataTypeMap.get(column);
+      AggregationResultHolder resultHolder = AggregationResultHolderFactory.getAggregationResultHolder(functionType, dataType);
       switch (functionType) {
       case MAX:
-        resultHolder = new DoubleAggregationResultHolder(dictionary.getDoubleValue(dictionary.length() - 1));
+        if (dataType.equals(FieldSpec.DataType.STRING)) {
+          resultHolder.setValue(dictionary.getStringValue(dictionary.length() - 1));
+        } else {
+          resultHolder.setValue(dictionary.getDoubleValue(dictionary.length() - 1));
+        }
         break;
       case MIN:
-        resultHolder = new DoubleAggregationResultHolder(dictionary.getDoubleValue(0));
+        if (dataType.equals(FieldSpec.DataType.STRING)) {
```

Comment:
Actually, seeing the pattern, you may want to factor this out as a reusable function.